contrib_bots/lib/giphy.py

```diff
@@ -0,0 +1,52 @@
+import requests
+
+
+class GiphyHandler(object):
+    '''
+    This plugin facilitates giving a gif image that appropriate with the
+    keywords user input. The gif image and search algorithm are provided by
+    giphy. It looks for messages starting with "@giphy", then we send an gif
+    image that appropriate with the keywords.
+    '''
+
+    GIPHY_API_KEY = 'dc6zaTOxFJmzC'  # This is public giphy API key.
```

Comment:
This is public beta key, right? You probably don't want to use it if you want to deploy to production, mainly because Giphy demands it that way. So, there needs to be a way to specify the API key.